implementations/rust/ockam/ockam_command/src/project/util.rs

```diff
@@ -174,91 +176,137 @@ pub async fn check_project_readiness<'a>(
+
+            // Handle the project show request result
+            // so we can provide better errors in the case orchestrator does not respond timely
+            if let Ok(_) = rpc
+                .request(api::project::show(&project_id, cloud_route))
+                .await
+            {
+                let p = rpc.parse_response::<Project>()?;
+                if p.is_ready() {
+                    println!("✅");
+                    return Ok(p.to_owned());
+                }
             }
-        }
+            print!(".");
+            Err(anyhow!("Project creation timed out. Plaese try again."))
+        })
+        .await?;
+
+        println!();
     }
-    if !project.is_reachable().await? {
+
+    {
         print!("Establishing connection (this can take a few minutes)...");
-        loop {
-            print!(".");
+        Retry::spawn(retry_strategy.clone(), || async {
             std::io::stdout().flush()?;
-            tokio::time::sleep(std::time::Duration::from_secs(10)).await;
-            if project.is_reachable().await? {
-                println!();
-                break;
+
+            // Handle the reachable result, so we can provide better errors in the case a project isn't
+            if let Ok(reachable) = project.is_reachable().await {
+                if reachable {
+                    println!("✅");
+                    return Ok(());
+                }
             }
-        }
+
+            print!(".");
+            Err(anyhow!("Timed out while trying to establish a connection to the project. Please try again."))
+        }).await?;
+
+        println!();
     }
+
     {
         print!("Establishing secure channel...");
         std::io::stdout().flush()?;
+
         let project_route = project.access_route()?;
         let project_identity = project
             .identity
             .as_ref()
-            .context("We already checked that the project has an identity")?
+            .context("Project identity is not set.")?
             .to_string();
-        match create_secure_channel_to_project(
-            ctx,
-            opts,
-            api_node,
-            tcp,
-            &project_route,
-            &project_identity,
-            CredentialExchangeMode::None,
-            None,
-        )
-        .await
-        {
-            Ok(sc_addr) => {
+
+        Retry::spawn(retry_strategy.clone(), || async {
+            std::io::stdout().flush()?;
+            if let Ok(sc_addr) = create_secure_channel_to_project(
+                ctx,
+                opts,
+                api_node,
+                tcp,
+                &project_route,
+                &project_identity,
+                CredentialExchangeMode::None,
+                None,
+            )
+            .await
+            {
                 // Try to delete secure channel, ignore result.
                 let _ = delete_secure_channel(ctx, opts, api_node, tcp, &sc_addr).await;
+                println!("✅");
```

Comment:
@CAOakleyII could you change the emojis to `✔︎`. We don't use emojis anywhere else in the interface so far, so let's make that change more consistently if decide its a good option.

Reply:
Absolutely!

Reply:
I wanted some sort of feedback that it worked, took a guess on the emoji, but agreed it feels out of place until a style is decided upon.

Reply:
great call on - we need some feedback there 👍